firebase-firestore/src/commonMain/kotlin/dev/gitlive/firebase/firestore/FirebaseTimestampSerializer.kt

```diff
@@ -0,0 +1,38 @@
+package dev.gitlive.firebase.firestore
+
+import dev.gitlive.firebase.FirebaseCompositeDecoder
+import dev.gitlive.firebase.FirebaseCompositeEncoder
+import kotlinx.serialization.Decoder
+import kotlinx.serialization.Encoder
+import kotlinx.serialization.KSerializer
+import kotlinx.serialization.SerialDescriptor
+import kotlinx.serialization.StructureKind
+
+class FirebaseTimestampSerializer : KSerializer<Timestamp> {
+
+    override val descriptor = object : SerialDescriptor {
+        val keys = listOf("seconds", "nanoseconds")
+        override val kind = StructureKind.OBJECT
+        override val serialName = "Timestamp"
+        override val elementsCount get() = 2
+        override fun getElementIndex(name: String) = keys.indexOf(name)
+        override fun getElementName(index: Int) = keys[index]
+        override fun getElementAnnotations(index: Int) = emptyList<Annotation>()
+        override fun getElementDescriptor(index: Int) = throw NotImplementedError()
+        override fun isElementOptional(index: Int) = false
+    }
```

Comment:
The `descriptor` is the same for both classes (Nullable or not), a super type could hold this common value..
For what regards the method the super type could accept a type `abstract class FirebaseSerializerSuper <T>(...)` and declare methods like `abstract fun serialize(encoder: Encoder, value: T)` then when extending the super pass either `Timestamp` or `Timestamp?`

Reply:
Good suggest, look 1da3b16